### PR TITLE
Additions for Modularity.

### PR DESCRIPTION
**Pull request by ralphbean**
This adds a new table and endpoint for storing metadata about built modules
that are not necessarily associated with a release of a distro or a product.

It makes PDC the canonical place for storing metadata about built modules which
may later be composed into products.  Tracking those composed combinations will
be in a different, future pull request.

### CODE REVIEW

**Comment by ralphbean** [comment]
See discussion in #407 for some background.

**Comment by simozhan** [comment]
@ralphbean looks there are some flake8 check errors need to be fix first

**Comment by ralphbean** [comment]
> there are some flake8 check errors need to be fix first

Thanks!  Those should be resolved now.

File: requirements/devel.txt [hunk 1]
```diff
@@ -14,3 +14,4 @@ lxml
 pygraphviz
 # docs
 Sphinx
+django-jsonfield
```
Comment (simozhan) [inline requirements/devel.txt]: Seems you have already added it in production.txt, maybe duplicate?

Comment (simozhan) [inline requirements/devel.txt]: Seems this has been added in production.txt, duplicate?

**Comment by simozhan** [comment]
I reviewed the codes, but not in detail.
Just some minor things like rebase on our latest codes to resolve conflicts maybe and fix the travis build issue, maybe the travis build issue has been already fixed in patches:
https://github.com/product-definition-center/product-definition-center/commit/a8695efe3e61d41dc59d031206bfc65fd75c08a0
https://github.com/product-definition-center/product-definition-center/commit/1b275224fceecd7d21bbe656d11afe6408018026

After fix this and resubmit it, I think we can first merge these commits. And if there is any bugs, let's create the issues and resolve it with a bug fix.
My idea is:
1. These codes have been already reviewed and deployed in fedora instance, it is qualified.
2. It hasn't break any existing api(seems release adds two new fields in optional, i guess it will not break anything)
Any concerns? ideas?

**Comment by simozhan** [comment]
@ralphbean Is that ok that you rebase on latest master codes and resubmit it again? So that we can merge it and ask QE people to have a test internally

**Comment by ralphbean** [comment]
> Is that ok that you rebase on latest master codes and resubmit it again?

Acknowledged.

We've been slammed the last week with some unplanned work and haven't gotten a chance to get back to this.

It's in our JIRA backlog as FACTORY-556 - we likely won't get to it until next sprint.

**Comment by simozhan** [comment]
@ralphbean Thanks a lot. Got your plan. No problem for us.

**Comment by sochotnicky** [comment]
FYI @mprahl  will be looking at rebasing/fixing up this PR

**Comment by mprahl** [comment]
@simozhan I rebased and fixed the Flake8 errors. Can you please review again?

**Comment by simozhan** [comment]
Thanks @sochotnicky for you info.
@mprahl thanks. LGTM :+1:

**Comment by ralphbean** [comment]
:beers:

**Pull request merged**